ts/features/payments/payment/components/WalletPaymentHeader.tsx

```diff
@@ -38,8 +45,13 @@ const WalletPaymentHeader = ({ currentStep }: WalletPaymentHeaderProps) => {
       return navigation.goBack();
     }
 
-    dispatch(walletPaymentSetCurrentStep(currentStep - 1));
-  }, [navigation, dispatch, goBackHandler, currentStep]);
+    if (currentStep === 3 && pspList.length === 1) {
+      dispatch(walletPaymentSetCurrentStep(currentStep - 2));
+      dispatch(walletPaymentResetPspList());
+    } else {
+      dispatch(walletPaymentSetCurrentStep(currentStep - 1));
+    }
+  }, [navigation, dispatch, goBackHandler, currentStep, pspList]);
```

Comment:
This could cause problems if any of the screens change position in the future. What do you think instead of defining constant indices for each step? Something like the following example:

```suggestion
    if (currentStep === PAYMENT_STEP_CONFIRM && pspList.length === 1) {
      dispatch(walletPaymentSetCurrentStep(PAYMENT_STEP_METHOD));
      dispatch(walletPaymentResetPspList());
    } else {
      dispatch(walletPaymentSetCurrentStep(PAYMENT_STEP_PSP));
    }
  }, [navigation, dispatch, goBackHandler, currentStep, pspList]);
```

Reply:
Great idea! But instead of using a constant, I think that using an enum could be more flexible and reusable as addressed in f2fb4130b54affb3d04659b731d7c59b24d0d798

What do you think?